### PR TITLE
Generate preference skeleton code from spree_cmd extension

### PR DESCRIPTION
**Pull request by j1mr10rd4n**
I found myself using [Spree Preferences](https://guides.spreecommerce.com/developer/preferences.html) often enough in my extensions that I added some skeleton code to the `spree_cmd` extension templates. Maybe others would find these useful as well.

The following options have been added to `spree extension`:

`--preferences`
Generates a namespaced configuration class for the extension. `engine.rb` initializes this and assigns it to a constant for use in the application.

`--preference-pane`
As above plus a backend page and controller for editing/updating the extension's preferences. A sub-menu is appended to the `Configuration` menu.

The extension tutorial has been updated to describe these new options.

Before adding this new section to the extension tutorial, I added some improvements (well I think they are!) [0381159](https://github.com/spree/spree/commit/0381159ad9567f38a2462b1d596f65a707a77917) encapsulates these.

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://j1mr10rd4n-spree-cmd-extension-preferences.spree-spree-9ed724b77adec04c49c2.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://j1mr10rd4n-spree-cmd-extension-preferences.spree-spree-9ed724b77adec04c49c2.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by JDutil** [comment]
Cool :+1: would be nice if we had a spec that executed the generator right now we manually run it before releases, and we may forget to run it with preference options.

File: cmd/lib/spree_cmd/templates/extension/lib/spree/%unprefixed_file_name%.rb.tt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+require 'spree_core'
+
+module Spree
+  module <%= unprefixed_class_name %>
+  end
+end
+
+require 'spree/<%= unprefixed_file_name  %>/engine'
+
+
```
Comment (BenMorganIO) [inline cmd/lib/spree_cmd/templates/extension/lib/spree/%unprefixed_file_name%.rb.tt]: Some extra line endings here.

**Comment by BenMorganIO** [comment]
I'm :-1: on this. Preferences are causing performance issues with Spree and I don't think there should be skeleton code encouraging this. I would rather opt for module attribute accessors such as how Devise does it.

**Comment by JDutil** [comment]
Good point @BenMorganIO I do think this could be confusing to newer developers, and encourage them to use preferences when they probably shouldn't be.

**Pull request closed without merging**